pyo3-macros-backend/src/pyclass.rs

```diff
@@ -383,242 +371,127 @@ fn impl_class(
-                visitor(collector.object_protocol_methods());
-                visitor(collector.async_protocol_methods());
-                visitor(collector.context_protocol_methods());
-                visitor(collector.descr_protocol_methods());
-                visitor(collector.mapping_protocol_methods());
-                visitor(collector.number_protocol_methods());
-            }
-            fn get_new() -> ::std::option::Option<::pyo3::ffi::newfunc> {
-                use ::pyo3::class::impl_::*;
-                let collector = PyClassImplCollector::<Self>::new();
-                collector.new_impl()
-            }
-            fn get_alloc() -> ::std::option::Option<::pyo3::ffi::allocfunc> {
-                use ::pyo3::class::impl_::*;
-                let collector = PyClassImplCollector::<Self>::new();
-                collector.alloc_impl()
-            }
-            fn get_free() -> ::std::option::Option<::pyo3::ffi::freefunc> {
-                use ::pyo3::class::impl_::*;
-                let collector = PyClassImplCollector::<Self>::new();
-                collector.free_impl()
-            }
-
-            fn for_each_proto_slot(visitor: &mut dyn ::std::ops::FnMut(&[::pyo3::ffi::PyType_Slot])) {
-                // Implementation which uses dtolnay specialization to load all slots.
-                use ::pyo3::class::impl_::*;
-                let collector = PyClassImplCollector::<Self>::new();
-                visitor(collector.object_protocol_slots());
-                visitor(collector.number_protocol_slots());
-                visitor(collector.iter_protocol_slots());
-                visitor(collector.gc_protocol_slots());
-                visitor(collector.descr_protocol_slots());
-                visitor(collector.mapping_protocol_slots());
-                visitor(collector.sequence_protocol_slots());
-                visitor(collector.async_protocol_slots());
-                visitor(collector.buffer_protocol_slots());
-                #methods_protos
-            }
-
-            fn get_buffer() -> ::std::option::Option<&'static ::pyo3::class::impl_::PyBufferProcs> {
-                use ::pyo3::class::impl_::*;
-                let collector = PyClassImplCollector::<Self>::new();
-                collector.buffer_procs()
+            fn py_class_descriptors(self) -> &'static [::pyo3::class::methods::PyMethodDefType] {
+                static METHODS: &[::pyo3::class::methods::PyMethodDefType] = &[#(#py_methods),*];
+                METHODS
             }
         }
+    }
+}
 
-        #alloc
-
-        #descriptors
-
-        #gc_impl
-    })
+fn extract_variant_data(variant: &syn::Variant) -> syn::Result<VariantPyO3> {
+    use syn::Fields;
+    let ident = match variant.fields {
+        Fields::Unit => &variant.ident,
+        _ => {
+            return Err(syn::Error::new(
+                variant.span(),
+                "Currently only support unit variants.",
+            ))
+        }
```

Comment:
There's a `bail_spanned!` macro which can help with boilerplate for this kind of stuff

```suggestion
        _ => bail_spanned!(variant.span() => "Currently only support unit variants.")
```